### DISCUSSION TITLE
WinUI project running Bg Tasks ApplicationTrigger

### DISCUSSION THREAD

**Discussion by danilobx** [microsoft/CsWinRT#1296] [Q&A]
Hi all!

I'm creating a solution with a simple WinUI project to trigger with an ApplicationTrigger, a Background Task in another WinRT project but without success. All the implementation was made using C# language.

Would it be possible to build a project as described? I managed to make it work using UWP.
Is there any example project similar to the one described?

Thanks!

**Comment by manodasanW** [comment]
I don't think we have a specific sample for this currently, but do you happen to have a specific issue you are running into or is it just the trigger doesn't run the background task?

**Reply by danilobx** [reply]
Hi Manodasan, thanks for your response!

Initially, I'm trying to build a WinUI project based on [BgTaskComponent](https://github.com/microsoft/CsWinRT/tree/master/src/Samples/BgTaskComponent) Sample. I uploaded my project into [BgTaskTest](https://github.com/danilobx/BgTaskTest/tree/main/BgTaskTest).

**First problem:** 
`Severity	Code	Description	Project	File	Line	Suppression State
Error	APPX1101	Payload contains two or more files with the same destination path 'WinRT.Host.runtimeconfig.json'. Source files: 
C:\CSharp\repo\BgTaskTest\BgTaskTest\BgTaskComponent\WinRT.Host.runtimeconfig.json
C:\CSharp\repo\BgTaskTest\BgTaskTest\BgTaskComponent\bin\x64\Debug\net6.0-windows10.0.19041.0\WinRT.Host.runtimeconfig.json	WinUIBgTaskUi	C:\Users\dls2\.nuget\packages\microsoft.windowsappsdk\1.2.221109.1\buildTransitive\Microsoft.Build.Msix.Packaging.targets	1504	
`
I did a workaround renaming the file in the project BgTaskComponent root to WinRT.Host.runtimeconfig2.json. Now, the application is running.

**Second problem:**
The application registers the defined Background Task, but when triggered by the Lifecycle events, nothing is happening. In this scenario, a notification would be displayed, like the original sample.
 
I don't know what is happening because it is not showing any error message, log, or exception.

**Comment by hez2010** [comment]
I came into the same issue.
BackgroundTask generally works except when you want to use `ApplicationTrigger`, where the background task won't be triggered by an `ApplicationTrigger` at all.

**Comment by Osirisoo0O** [comment]
Any new progress？
I got “_COMPlusExceptionCode = -532462766” when register background task with ApplicationTrigger.
